spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/msi/AzureConfigMSIConnector.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.config.msi;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
+
+import java.util.Optional;
+
+import static com.microsoft.azure.spring.cloud.config.msi.ConfigAccessKeyResource.ARM_ENDPONT;
+
+/**
+ * Get connection string for configured config store from ARM with MSI access token.
+ */
+public class AzureConfigMSIConnector {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureConfigMSIConnector.class);
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private static final CloseableHttpClient HTTP_CLIENT = HttpClients.createDefault();
+    private final ConfigMSICredentials msiCredentials;
+    private final ConfigAccessKeyResource keyResource;
+
+    public AzureConfigMSIConnector(ConfigMSICredentials credentials, ConfigAccessKeyResource keyResource) {
+        this.msiCredentials = credentials;
+        this.keyResource = keyResource;
+    }
+
+    public String getConnection() {
```

Comment:
`getConnection` is confusing here.

Reply:
Will change to `getConnectionString`, hope it's more clear.